docs/guides/sharing-npm-dependencies.md

```diff
@@ -25,19 +40,122 @@ const {featureAppManager} = createFeatureHub('acme:integrator', {
+
+### Integrator
+
+When using the
+[`@feature-hub/module-loader-federation`][module-loader-federation-api] package,
+the integrator's client module must be bundled with Webpack. The integrator can
+share npm dependencies to Feature Apps using the `shared` property of its
+`ModuleFederationPlugin`:
+
+```js
+module.exports = {
+  // ...
+  plugins: [
+    new webpack.container.ModuleFederationPlugin({
+      shared: {
+        react: {singleton: true, eager: true},
+      },
+    }),
+  ],
+};
+```
+
+If the integrator wants to provide a shared module:
+
+- It must be `eager` if the integrator itself consumes it synchronously.
+- It does _not_ need to be `eager` if the integrator consumes it asynchronously
+  (dynamic import).
+- It must be a `singleton` if the module is stateful.
+- It must be a `singleton` if the module can not exist more than once on a page.
+
+There is no need for the integrator to share a module that it does not consume
+itself. In this case, those Feature Apps that need it can provide the shared
+module instead (see below).
+
+### Feature Apps
+
+If Feature Apps want to use shared dependencies, they must define them in the
+`shared` property of their `ModuleFederationPlugin`.
+
+```js
+module.exports = {
+  // ...
+  plugins: [
+    new webpack.container.ModuleFederationPlugin({
+      // ...
+      shared: {
+        react: {singleton: true},
+      },
+    }),
+  ],
+};
+```
+
+A dependency that the integrator has provided as `singleton` must also be a
+`singleton` in the Feature App's Webpack config. But here, it should _not_ be
+`eager`, otherwise it will always be loaded, regardless of whether the
+integrator has already provided it. Version mismatches of singletons are
+generally ignored, and a warning is logged, unless `strictVersion` is set to
+`true`, in which case a runtime error is thrown.
+
+The biggest benefit of using the Module Federation Loader over the AMD Module
+Loader (or in addition), is that Feature Apps can also share npm dependencies
+with each other, without the need to involve the integrator.
+
+```js
+module.exports = {
+  // ...
+  plugins: [
+    new webpack.container.ModuleFederationPlugin({
+      // ...
+      shared: {
+        '@apollo/client': {},
+      },
+    }),
+  ],
+};
 ```
 
-On the server, when Feature Apps are [bundled as CommonJS modules][serversrc],
-the integrator either
+If a Feature App wants to provide a shared module, it should _not_ be `eager`.
+In case of a version mismatch (defined by the [semver][] ranges in the
+`package.json` files of the Feature Apps) the module is loaded multiple times,
+and available to other Feature Apps in each loaded version.
+
+> **Note**:  
+> A large library of which only a small portion is used, might not be a good
+> candidate to provide as a shared module. In this case it might be more
+> efficient to include the library in each Feature App bundle, while making use
+> of [Webpack's tree shaking capabilities][tree-shaking].
```

Comment:
Maybe we could add here: It is more efficient to share a custom wrapper of a library providing a common set of use cases for the feature apps. Thus, the bundle size can be reduced by tree shaking.

Reply:
we could also talk about the approach taken by OneShop to load the library lazily. This is done for feature services that are only used in the client. It should be possible for client side rendered feature apps also.

Reply:
Should we maybe add these details to the "Reducing the Bundle Size" guide and cross reference it from here?